service/matching/tasklist/matcher_test.go

```diff
@@ -579,6 +563,18 @@ func (t *MatcherTestSuite) TestIsolationMustOfferRemoteMatch() {
 	t.Equal(t.taskList.Parent(20), req.GetTaskList().GetName())
 }
 
+func (t *MatcherTestSuite) TestPollersDisconnectedAfterDisconnectBlockedPollers() {
```

Comment:
It would be great to also simulate the scenario where tasklist ownership changes and this change reduces task latencies by preventing hanging polls. Simulation framework currently doesn't support such ownership change but should be straightforward to introduce

Reply:
Manual local testing is good if you know what you are doing but having it defined as another simulation framework would be preferred. We will run those simulation scenarios as part of CI and ensure features/improvements like this are not broken going forward. 
Not a blocker for this PR. Let's add it when we have cycles.